src/Agent.Listener/DistributedTask.Pipelines/Yaml/TypeConverters/ConverterUtil.steps.cs

```diff
@@ -0,0 +1,622 @@
+                        case YamlConstants.FailOnStderr:
+                            task.Inputs["failOnStderr"] = parser.Expect<Scalar>().Value ?? String.Empty;
+                            break;
+                        case YamlConstants.WorkingDirectory:
+                            task.Inputs["workingDirectory"] = parser.Expect<Scalar>().Value ?? String.Empty;
+                            break;
+                        default:
+                            SetTaskControlProperty(parser, task, scalar);
+                            break;
+                    }
+                }
+
+                result = task;
+            }
+            else if (String.Equals(scalar.Value, YamlConstants.PowerShell, StringComparison.Ordinal))
+            {
+                var task = new TaskStep
+                {
+                    Enabled = true,
+                    Reference = new TaskReference
+                    {
+                        Name = "PowerShell",
+                        Version = "2",
+                    },
+                    Inputs = new Dictionary<String, String>(StringComparer.OrdinalIgnoreCase),
+                };
+
+                task.Inputs["targetType"] = "inline";
+                task.Inputs["script"] = parser.Expect<Scalar>().Value ?? String.Empty;
+                while (parser.Allow<MappingEnd>() == null)
+                {
+                    scalar = parser.Expect<Scalar>();
+                    switch (scalar.Value ?? String.Empty)
+                    {
+                        case YamlConstants.ErrorActionPreference:
+                            task.Inputs["errorActionPreference"] = parser.Expect<Scalar>().Value ?? String.Empty;
+                            break;
+                        case YamlConstants.FailOnStderr:
+                            task.Inputs["failOnStderr"] = parser.Expect<Scalar>().Value ?? String.Empty;
+                            break;
+                        case YamlConstants.IgnoreLASTEXITCODE:
+                            task.Inputs["ignoreLASTEXITCODE"] = parser.Expect<Scalar>().Value ?? String.Empty;
+                            break;
+                        case YamlConstants.WorkingDirectory:
+                            task.Inputs["workingDirectory"] = parser.Expect<Scalar>().Value ?? String.Empty;
+                            break;
+                        default:
+                            SetTaskControlProperty(parser, task, scalar);
+                            break;
+                    }
+                }
+
+                result = task;
+            }
+            else if (String.Equals(scalar.Value, YamlConstants.Checkout, StringComparison.Ordinal))
+            {
+                var checkoutStep = new CheckoutStep();
+                scalar = parser.Expect<Scalar>();
+                checkoutStep.Name = scalar.Value ?? String.Empty;
+                if (String.Equals(checkoutStep.Name, YamlConstants.Self, StringComparison.Ordinal))
```

Comment:
`- checkout: self` supports `clean`, `fetchDepth`, and `lfs` since the agent understand variables to influence those settings. I don't want to figure out how to merge the settings into the resource instance for the job, since ultimately that's the wrong pattern.

The approach I've taken here, does have a limitation. Whether to checkout submodules and nested submodules can't be influence today via an agent env var.